src/main/java/seedu/exercise/logic/commands/statistic/StatsFactory.java

```diff
@@ -114,50 +119,51 @@ private Statistic generatePieChartStatistic() {
 
         ArrayList<String> names = hashMapNameToList(data);
         ArrayList<Double> values = hashMapDoubleToList(data, names);
+        double total = getTotal(values);
+        double average = getAverage(total);
 
-        return new Statistic(category, chart, startDate, endDate, names, values);
+        return new Statistic(category, chart, startDate, endDate, names, values, total, average);
     }
 
     /**
-     * Compute exercise count with filtered exercises list.
+     * Returns the sum of all values.
+     * Values can be calories or exercise frequency.
      */
-    private HashMap<String, Double> getTotalExerciseFrequency() {
-        ArrayList<Exercise> filteredExercise = getFilteredExercise();
-        HashMap<String, Double> data = new HashMap<>();
-
-        for (Exercise e : filteredExercise) {
-            Name name = e.getName();
-            Unit unit = e.getUnit();
-            String nameWithUnit = name.toString() + " (" + unit.toString() + ")";
-            double count = 1;
-
-            if (data.containsKey(nameWithUnit)) {
-                count = data.get(nameWithUnit) + 1;
-            }
-
-            data.put(nameWithUnit, count);
+    private double getTotal(ArrayList<Double> values) {
```

Comment:
You can actually remove the second line of documentation since this is such a general function of just summing up all the values in an arraylist

Reply:
ok